docs/src/explanations/chisel-enum.md

```diff
@@ -0,0 +1,131 @@
+# ChiselEnum
+
+The ChiselEnum type is an easy way to get away from value encodings and reduce the chance of error when encoding muxes, opcodes, and functional unit operations. As opposed to the Chisel type Enums, ChiselEnums allow for names to be explicitly passed through Bundles like IO. 
+
+## Importing the ChiselEnum module
+
+```scala mdoc
+import chisel3._
+import chisel3.util._
+import chisel3.stage.ChiselStage
+import chisel3.experimental.ChiselEnum
+
+```
+
+## Functionality and Examples
+
+Below we see ChiselEnum being used as mux select for a RISC-V core. While wrapping the object in a package is not required, it is highly recommended as it allows for the type to be used in multiple files more easily. 
+
+```scala mdoc
+// package CPUTypes {
+    object AluMux1Sel extends ChiselEnum {
+        val rs1out, pcout = Value
+        /**
+            "rs1out" -> 0.U,
+            "pcout"  -> 1.U
+        */
+    }
+// }
+```
+
+Here we see a mux using the AluMux1Sel to select between different inputs. 
+
+```scala mdoc
+import AluMux1Sel._
+
+class AluMux1File extends Module {
+    val io = IO(new Bundle {
+        val aluMux1Sel =  Input( AluMux1Sel() )
+        val rs1Out     =  Input(Bits(32.W))
+        val pcOut      =  Input(Bits(32.W))
+        val aluMux1Out = Output(Bits(32.W))
+    })
+
+    // Default value for aluMux1Out
+    io.aluMux1Out := 0.U
+
+    switch (io.aluMux1Sel) {
+        is (rs1out) {
+            io.aluMux1Out  := io.rs1Out
+        }
+        is (pcout) {
+            io.aluMux1Out  := io.pcOut
+        }
+    }
+}
+```
+```scala mdoc:verilog
+ChiselStage.emitVerilog(new AluMux1File )
+```
+
+ChiselEnum also allows for the user to define variables by passing in the value shown below. Note that the value must be increasing or else 
+ > chisel3.internal.ChiselException: Exception thrown when elaborating ChiselGeneratorAnnotation
+is thrown during Verliog generation.
```

Comment:
something weird here in the formatting. You can actually make the `mdoc` just display the thrown error by doing the bad thing in the code

Reply:
I added space around the block quote for now. I'm worried that the ci will start going crazy if I intentionally throw errors in the mdoc

Reply:
For an expected exception, I think `scala mdoc:crash` is what you want. There's also `scala mdoc:fail`. Lots of documentation on this [here](https://scalameta.org/mdoc/docs/modifiers.html).